packages/utils/src/diff/diffStrings.ts

```diff
@@ -7,8 +7,24 @@
 
 import * as diff from 'diff-sequences'
 import { DIFF_DELETE, DIFF_EQUAL, DIFF_INSERT, Diff } from './cleanupSemantic'
+import type { DiffOptions } from './types'
+
+function diffStrings(a: string, b: string, options?: DiffOptions): [Array<Diff>, boolean] {
+  const truncate = options?.truncateThreshold ?? false
+  const truncateThreshold = Math.max(Math.floor(options?.truncateThreshold ?? 0), 0)
+  let aLength = a.length
+  let bLength = b.length
+  if (truncate) {
+    const aMultipleLines = a.includes('\n')
+    const bMultipleLines = b.includes('\n')
+    // multiple-lines string expects a newline to be appended at the end
+    const _a = aMultipleLines ? `${a.split('\n').slice(0, truncateThreshold).join('\n')}\n` : a
```

Comment:
The new line is `\r\n` on Windows. First, we need to get what the new line symbol is used here (it's still possible to have `\n` on Windows, so we need to check the string itself):

```js
function getNewLineSymbol(string) {
  return string.includes('\r\n') ? '\r\n' : '\n'
}
```

We can also limit the number of elements since we know how many we need. This should be more performant on large strings (which is what we are targeting in this PR)

```suggestion
    const _a = aMultipleLines ? `${a.split(newLineSymbol, truncateThreshold).slice(0, truncateThreshold).join(newLineSymbol)}\n` : a
```

Reply:
Thanks for the feedback, PR adjusted!